File: algorithms/cpp/largestMergeOfTwoStrings/largestMergeOfTwoStrings.cpp
// Source: https://leetcode.com/problems/largest-merge-of-two-strings/
// Author: Miao Zhang
// Date:   2021-06-03

class Solution {
public:
    string largestMerge(string word1, string word2) {
        string res;
        int w1 = word1.length(), w2 = word2.length();
        int i = 0, j = 0;
        while (i < w1 && j < w2) {
            res += word1.substr(i) > word2.substr(j) ? word1[i++] : word2[j++];
        }
        res.append(word1.substr(i));
        res.append(word2.substr(j));
        return res;
    }
};

File: algorithms/python/largestMergeOfTwoStrings/largestMergeOfTwoStrings.py
#! /usr/bin/env python3
# -*- coding: utf-8 -*-

# Source: https://leetcode.com/problems/largest-merge-of-two-strings/
# Author: Miao Zhang
# Date:   2021-06-03

class Solution:
    def largestMerge(self, word1: str, word2: str) -> str:
        res = ''
        w1 = len(word1)
        w2 = len(word2)
        i, j = 0, 0
        while i < w1 and j < w2:
            if word1[i:] > word2[j:]:
                res += word1[i]
                i += 1
            else:
                res += word2[j]
                j += 1
        res += word1[i:]
        res += word2[j:]
        return res
